sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1543,6 +1544,90 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResults<Application>> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "currentAssessment.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "application.applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest",
+      )
+      .innerJoin("application.student", "student")
```

Comment:
Should this be a left or an inner join? Since we only need the students for the application and not all the students in my understanding.